matrix_client/api.py

```diff
@@ -164,13 +164,22 @@ def logout(self):
         """
         return self._send("POST", "/logout")
 
-    def create_room(self, alias=None, is_public=False, invitees=None):
+    def create_room(
+                self,
+                alias=None,
+                name=None,
+                is_public=False,
+                invitees=None,
+                federate=None
```

Comment:
I don't like leaving booleans as ``None``. You should set this to true, so that users of the API know that federation is on by default. I'd actually argue ``None`` is closer to ``False``, which is obviously dangerously wrong. 

https://matrix.org/docs/spec/client_server/unstable.html#m-room-create

Reply:
On my homeservet it's off by default...
I prefer to give a choice of three options: explicitly turn on, explicitly turn off and leave by default.

Reply:
The default given in the spec is ``True``. If your homeserver does something different then it's non-conforming?

Reply:
We discuss about SDK implementation or about settings on my homeserver? I gave an example to show that the behavior of the server is not always obvious and it is useful to give the developer the opportunity to explicitly set this parameter.

Reply:
We should specify the matrix spec default in the methods docstring. `None` is fine as default kwarg; we just need to be explicit that according to matrix spec this defaults to `True` in the docstring. If servers have a setting that can change the default value of this and that's considered part of the spec, it's a spec bug that the setting isn't mentioned at the place @Half-Shot linked.

Reply:
So based on a conversation with @ara4n in #matrix-spec:matrix.org, a server that had a default value other than `True` for room federation would be non-compliant with the matrix spec. I'm fine with including this distinction between `None` and `False` if it's useful to you, but the docstring for this method needs to explicitly say that `True` is the default.